### PR TITLE
feat: onboard kubernetes-binaries, azure-acr-credential-provider and containerd-wasm-shims to renovate

### PR DESCRIPTION
**Pull request by Devinwong**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Onboard `kubernetes-binaries`, `azure-acr-credential-provider` and `containerd-wasm-shims` to renovate. Moved them from acs-mirror to MAR and downloaded by Oras.
Tested in my fork but still need to merge to validate.
All e2e tests passed, except the GPU tests which didn't have GPU quota.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: .github/README.md [hunk 1]
```diff
@@ -1,9 +0,0 @@
-## AgentBaker GitHub Repository
```
Comment (Devinwong) [inline .github/README.md:1]: Removed this readme as if it will be the default readme shown on AgentBaker root. We already have a readme at root level. We also have a readme for renovate in this folder.
I (Devin) created this readme so it's safe to remove.

File: CODEOWNERS [hunk 2]
```diff
@@ -1,7 +1,4 @@
 * @juan-lee @cameronmeissner @UtheMan @ganeshkumarashok @anujmaheshwari1 @AlisonB319 @Devinwong @lilypan26 @ShiqianTao @AbelHu @junjiezhang1997 @jason1028kr @djsly @phealy @r2k1 @timmy-wright
 
-# Code owners for cse_cmd.sh.gtpl and cse_cmd.sh. 
-# In case someone modifies one file but forgets the other one.
-# Testing if it will work as expected.
-cse_cmd.sh.gtpl @Devinwong @lilypan26 @r2k1 @timmy-wright
```
Comment (Devinwong) [inline CODEOWNERS:6]: removing cse_cmd.sh.gtpl as we no longer has this file

File: pkg/proto/aksnodeconfig/v1/README.md [hunk 3]
```diff
@@ -113,7 +113,7 @@ Nevertheless, it’s not a big harm to use `optional` even though it’s not nee
 1. Set default values for your variables, if the existing defaulting provided by `proto3` doesn't fit your purpose. For example, if a bool variable is not set, `proto` will default it to `false`. However, if you want to default it to `true`, then you can set your own default function. `getDisableSSH` in `cse_cmd.sh.gtpl` is 1 example.
 
 ## Detailed steps with example
-Example: IMDSRescrtionConfig
+Example: IMDSRestrictionConfig [Example PR](https://github.com/Azure/AgentBaker/pull/5154)
```
Comment (djsly) [inline pkg/proto/aksnodeconfig/v1/README.md:116]: Should this be part of another PR ?

File: parts/linux/cloud-init/artifacts/components.json [hunk 4]
```diff
@@ -784,36 +784,36 @@
             "versionsV2": [
               {
                 "k8sVersion": "1.27",
-                "renovateTag": "<DO_NOT_UPDATE>",
-                "latestVersion": "1.27.101-akslts",
-                "previousLatestVersion": "1.27.100-akslts"
+                "renovateTag": "OCI_registry=https://mcr.microsoft.com, name=oss/binaries/kubernetes/kubernetes-node",
```
Comment (djsly) [inline parts/linux/cloud-init/artifacts/components.json:787]: What is OCI_registry ?

File: .github/README.md (hunk 1 above)
Comment (djsly) [inline .github/README.md:1]: Why removing this file ?  Since it was showing up as the main text on the front page ?

Comment (Devinwong) [reply .github/README.md:1]: Yes, it was showing up as the main text on the front page. It should show the readme in the root folder.

File: pkg/proto/aksnodeconfig/v1/README.md (hunk 3 above)
Comment (Devinwong) [reply pkg/proto/aksnodeconfig/v1/README.md:116]: ya. Just taking a free ride because it's just a readme change. 😅

File: .github/README.md (hunk 1 above)
Comment (Devinwong) [reply .github/README.md:1]: As mentioned in another comment
_Removed this readme as if it will be the default readme shown on AgentBaker root. We already have a readme at root level. We also have a readme for renovate in this folder.
I (Devin) created this readme so it's safe to remove._

File: parts/linux/cloud-init/artifacts/components.json (hunk 4 above)
Comment (Devinwong) [reply parts/linux/cloud-init/artifacts/components.json:787]: Setting a new keyword to distinguish from the containerImage one `registry`. OCI_registry here means it's a registry hosting OCI artifacts. I apply different package rule to this keyword.

**Review by djsly — APPROVED** [review]

File: parts/linux/cloud-init/artifacts/components.json (hunk 4 above)
Comment (Devinwong) [reply parts/linux/cloud-init/artifacts/components.json:787]: oh. You are right. I forgot to paste the OCI_registry custom manager from my fork. Thanks for pointing this out. Added it back
